### ISSUE TITLE
features normalization 

### ISSUE THREAD

**Issue by zhanghao731** [tim-learn/ATDOC#4]
Hi,thanks for sharing your code! I have a question about calculating the cosine distance in **demo_uda , line 267**. `dis = -torch.mm(features_target.detach(), mem_fea.t())`I think before calculating the cosine distance,the features should be normalized.May be I'm missing something?

**Comment by tim-learn** [comment]
> Hi,thanks for sharing your code! I have a question about calculating the cosine distance in **demo_uda , line 267**. `dis = -torch.mm(features_target.detach(), mem_fea.t())`I think before calculating the cosine distance,the features should be normalized.May be I'm missing something?

Using cosine distance and inner distance is the same since we want to find its neighbors from the normalized memory bank. For example, comparing <a,b> and <a,c>, the norm of a does not matter the comparison between cosine(a,b) and cosine(a,c).
